src/actions/attack.cpp

```diff
@@ -187,8 +187,9 @@ battle_context_unit_stats::battle_context_unit_stats(nonempty_unit_const_ptr up,
 	int damage_multiplier = 100;
 
 	// Time of day bonus.
+	unit_alignments::type alignment = weapon->alignment() ? *(weapon->alignment()) : u.alignment();
```

Comment:
Since this returns an `optional` now, you should use `weapon->alignment().value_or(u.alignment())`;